analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,186 @@
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar") 	#Need to set this since pathwayName and clinVarName will change which is not good for aov function
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95)
+	posthoc <- data.frame(posthoc[[1]])
+	aovOutNamed <- unlist(summary(aovOut))
+	posthoc[,"F_Value"] <- aovOutNamed["F value1"]
+	posthoc[,"AOV_P"] <- aovOutNamed["Pr(>F)1"]
+	posthoc[,"Pathway"] <- pathwayName
+	posthoc[,"ClinVar"] <- clinVarName
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc) #Pulls out disease names 
+	colnames(posthoc)[1:2] <- c("VarX", "VarY")
+	return(posthoc)
+}
+
+
+#Analysis by short_histology. This is only clinical variable we are examining but the function compareClinVarToPathway
+#is generic enough that any variable in clinical data can be used. This will be useful if we decide to use for instance molecular
+#subtypes instead of histology for this analysis. So the lapply below will for each pathway
+#run an ANOVA across all histologies i.e. looking to see if any particular histology has a significantly greater score
+#than any other histology and then follow that with a post-hoc test. The comparison is not done between pathways. 
+diseaseType <- lapply(rownames(geneSetExpMat), FUN=compareClinVarToPathway, clinVarName="short_histology")
+diseaseType <- do.call("rbind", diseaseType) 
+
+#Filter to significant entries by F and Tukey and write out 
+diseaseTypeFilt <- diseaseType %>% dplyr::filter( AOV_P <= aovParam, p.adj <= padjParam)
+colnames(diseaseTypeFilt)[1:2] <- c("DiseaseX", "DiseaseY") #Rename columns to match clinical variable type
+write.table(diseaseTypeFilt, "results/DiseaseCorrelationPathway.txt", sep="\t", row.names=F)
+
+#We would like to filter to cases where a histology is significantly higher than other histologies (and the corresponding pathways)
+#This is for the heatmap so we may focus a bit on the more interesting results
+#The issue here is that Tukey will write out the pathways in some particular order
+#If we are interested in finding if a Histology A is significantly upregulated compared to at least X% of
+#other histologies this may present a problem because Tukey may write out the following
+#HistologyA-HistologyB ~ Fold-change = 1
+#HistologyA-HistologyC ~ Fold-change = 1
+#HistologyD-HistologyA ~ Fold-change = -1
+#if the first histology is in column 1 and the second is in column 2 and the FC is in column 3 of a data frame we would be required
+#to check the FC in order to determine if Histology Y is in fact > or < another histology. Hence a bit of logic below to get counts. 
+diseaseTypeFilt[,"DiseaseTmpX"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"DiseaseX"]), as.character(diseaseTypeFilt[,"DiseaseY"]))
+diseaseTableTmp <- data.frame(table(diseaseTypeFilt[,c("DiseaseX", "Pathway")])) #This gets the counts per disease/pathway, i.e. number of other diseases it's greater than
```

Comment:
Should this be
```suggestion
diseaseTableTmp <- data.frame(table(diseaseTypeFilt[,c("DiseaseTmpX", "Pathway")])) #This gets the counts per disease/pathway, i.e. number of other diseases it's greater than
```
This is what I would expect based on the comment.

Reply:
If I look at `diseaseTableTmp` after line 116
```r
> head(diseaseTableTmp)
            DiseaseX                                    Pathway Freq
883       Schwannoma         HALLMARK_INTERFERON_ALPHA_RESPONSE   16
471     Neurofibroma HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION   15
475       Schwannoma HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION   15
1457    Neurofibroma                   HALLMARK_SPERMATOGENESIS   15
196  Medulloblastoma                    HALLMARK_APICAL_SURFACE   14
469       Meningioma HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION   14
```
And then look at the `Neurofibroma` and `HALLMARK_SPERMATOGENESIS` pair:

```r
> head(diseaseTypeFilt %>% dplyr::filter(DiseaseX == "Neurofibroma",
+                                        Pathway == "HALLMARK_SPERMATOGENESIS"))
      DiseaseX             DiseaseY       diff        lwr         upr        p.adj  F_Value
1 Neurofibroma              Adenoma -0.7098306 -1.2511922 -0.16846900 3.217160e-04 13.03797
2 Neurofibroma                 ATRT -0.4293781 -0.6789335 -0.17982274 4.055654e-08 13.03797
3 Neurofibroma  Central neurocytoma -0.6354246 -1.1767862 -0.09406301 3.734714e-03 13.03797
4 Neurofibroma Choroid plexus tumor -0.4566901 -0.7428506 -0.17052952 7.960704e-07 13.03797
5 Neurofibroma              CNS EFT -0.4602077 -0.8246229 -0.09579253 7.899308e-04 13.03797
6 Neurofibroma  CNS Embryonal tumor -0.5239934 -0.8150540 -0.23293291 5.442339e-09 13.03797
        AOV_P                  Pathway         ClinVar          DiseaseTmpX
1 7.64177e-58 HALLMARK_SPERMATOGENESIS short_histology              Adenoma
2 7.64177e-58 HALLMARK_SPERMATOGENESIS short_histology                 ATRT
3 7.64177e-58 HALLMARK_SPERMATOGENESIS short_histology  Central neurocytoma
4 7.64177e-58 HALLMARK_SPERMATOGENESIS short_histology Choroid plexus tumor
5 7.64177e-58 HALLMARK_SPERMATOGENESIS short_histology              CNS EFT
6 7.64177e-58 HALLMARK_SPERMATOGENESIS short_histology  CNS Embryonal tumor
```

`HALLMARK_SPERMATOGENESIS` looks to have lower expression in `Neurofibroma` compared to every other histology.

Reply:
That's absolutely correct. I think when I refactored the code a bit, I didn't update that part correctly. Thanks for spotting it, I am going to update it but as per your other suggestion, give it a more descriptive name.